flink-clients/src/main/java/org/apache/flink/client/program/PackagedProgram.java

```diff
@@ -218,12 +217,7 @@ public String getDescription() throws ProgramInvocationException {
      * local execution by default.
      */
     public void invokeInteractiveModeForExecution() throws ProgramInvocationException {
-        FlinkSecurityManager.monitorUserSystemExitForCurrentThread();
-        try {
-            callMainMethod(mainClass, args);
-        } finally {
-            FlinkSecurityManager.unmonitorUserSystemExitForCurrentThread();
-        }
+        callMainMethod(mainClass, args);
```

Comment:
The jira says it is to deprecate the FlinkSecurityManager, but the PR is removing it. It seems normal practise to deprecate prior to removing.

Reply:
The JDK Security Manager APIs were [deprecated in JDK 17](https://openjdk.org/jeps/411) which was released in 2021, and it has been [removed in JDK 24](https://openjdk.org/jeps/486). This code doesn't build with DJK 24 and newer.

I'm not sure there's a great way to keep this functionality around in a deprecated state and support newer JDKs. It might be possible to do something with multi-release jars.

I guess another option would be to mark `FlinkSecurityManager` as deprecated, and then remove it in a later release, and wait until that later release for Flink to support JDK 24 and newer.

Reply:
I tend to think such change deserves a dedicated discussion in mailing list first